build_test/lib/src/resolve_source.dart

```diff
@@ -193,14 +193,22 @@ Future<T> _resolveAssets<T>(
     sourceAssets: inputAssets,
     rootPackage: rootPackage,
   );
-  // We don't care about the results of this build.
-  unawaited(runBuilder(
-    resolveBuilder,
-    inputAssets.keys,
-    MultiAssetReader([inMemory, assetReader]),
-    InMemoryAssetWriter(),
-    resolvers ?? defaultResolvers,
-  ));
+  // We don't care about the results of this build, but we also can't await
+  // it because that would block on the `tearDown` of the `resolveBuilder`.
+  //
+  // We also dont want to leak unhandled async errors so we swallow them.
```

Comment:
I don't understand this part. If we swallow unhandled async errors I think we mask bugs in the test? Does it work if we do `unawaited(runBuilders(...).catchError((_) {}));`

Reply:
Oh right - I think that would work and be better :). The builder itself doesn't have unawaited futures in its build method (well unless the action does).